src/python/pants/backend/project_info/peek.py

```diff
@@ -162,21 +190,88 @@ def describe_ruleset(ruleset: DependencyRuleSet | None) -> tuple[str, ...] | Non
     return ruleset.peek()
 
 
+async def _create_target_alias_to_goals_map() -> dict[str, tuple[str, ...]]:
+    """Returns a mapping from a target alias to the goals that can operate on that target.
+
+    For instance, `pex_binary` would map to `("run", "package")`.
+
+    :return: A mapping from a target alias to the goals that can operate on that target.
+    """
+    # TODO: This is manually curated for now - do we want to show all possible goals?
+    peekable_field_sets: list[type[FieldSet]] = [
+        DeployFieldSet,
+        PackageFieldSet,
+        PublishFieldSet,
+        RunFieldSet,
+        TestFieldSet,
+    ]
+    # TODO: How do we get the goal subsystem from field set? This shouldn't need to be manually maintained
+    field_set_to_goal_map: dict[type[FieldSet], str] = {
+        DeployFieldSet: "experimental-deploy",
```

Comment:
Could these be advisory properties on the FieldSet subclasses? That way we could be sure they were being updated.

Reply:
I'm not sure - would that make sense? The way in which they're linked always felt de-coupled, for something that feels intrinsically coupled - I just didn't know if it was setup the way it is for a reason.

Reply:
Well, we're formalizing a new concept here, namely "goals you can run on this target". So maybe it makes sense to encode that more formally as well.

Reply:
If we want to do that, I don't really want to tackle that here. 

I'd rather experiment a bit with these 5 goals, then looking at formalizing something with the remainder. From my investigation into this, the goals I'm using - it would be trivial to do this, but for other goals, it could be messier.